libexec/rtld-elf/riscv/reloc.c

```diff
@@ -257,6 +311,12 @@ reloc_non_plt(Obj_Entry *obj, Obj_Entry *obj_rtld, int flags,
 	Elf_Addr *where;
 	unsigned long symnum;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+	/* The relocation for the dynamic loader has already been done. */
+	if (obj == obj_rtld)
+		return (0);
```

Comment:
In the debug RTLD it might still be useful to verify that there are no outstanding relocations in RTLD since that indicates a code generation/linker bug.

Reply:
I don't think that actually achieves anything. The most likely case will be that __cap_relocs had relocations needing to be applied to it, and so our attempts to construct the capabilities it wants will trap, i.e. before we get here.

Reply:
I guess we could add `objdump -r | wc -l` to the build to check we don't regress?

Reply:
That feels like a gross check that gets added, never actually catches anything and then years later you notice is still there...

Reply:
How about `assert(obj->relasize == 0 && "should not have elf relocations in rtld")`?